health/health.c

```diff
@@ -803,15 +802,25 @@ static void initialize_health(RRDHOST *host)
 
     long n = config_get_number(CONFIG_SECTION_HEALTH, "in memory max health log entries", host->health_log.max);
     if(n < 10) {
-        netdata_log_error("Host '%s': health configuration has invalid max log entries %ld. Using default %u",
-                          rrdhost_hostname(host),
-                          n,
-                          host->health_log.max);
+        log_health("Host '%s': health configuration has invalid max log entries %ld. Using default %u", rrdhost_hostname(host), n, host->health_log.max);
         config_set_number(CONFIG_SECTION_HEALTH, "in memory max health log entries", (long)host->health_log.max);
     }
     else
         host->health_log.max = (unsigned int)n;
 
+    //default health log history is 5 days and not less than a day
+    if (host->health_log.health_log_history < 86400) {
+        uint32_t m = config_get_number(CONFIG_SECTION_HEALTH, "health log history", HEALTH_LOG_DEFAULT_HISTORY);
```

Comment:
86400 should be a define -- e.g. HEALTH_LOG_MINIMUM_HISTORY

Now probably if minimum required is 86400 and there is nothing explicitly defined, it should use the default (HEALTH_LOG_DEFAULT_HISTORY) 

However if there is a value defined and it is less than 86400 it would make sense to switch to the minimum one (HEALTH_LOG_MINIMUM_HISTORY)

Reply:
I agree 100% with everything @stelfrag suggests, I am only waiting for commits to address them.